Widescreen patch
Seles, Commander cutscene
Knight of Sandora
--- original
+++ modified
@@ -486,14 +486,35 @@
 rel :JMP_a38_1
 rel :JMP_a38_2
 rel :JMP_a38_1
+#include ../../../../patches/libs/widescreen.txt
 JMP_a38_0:
 call 3, 0x19, stor[8]
 jmp_cmp !=, 0x0, stor[8], inl[:LABEL_23]
+mov 0xa, inl[:WS_WAIT_FLAG]
+mov 1, inl[:WS_WAIT_FLAG_MATCH]
+gosub inl[:WS_FLAG_WAIT]
+call 101, stor[0], 0xfffffe3e, 0xffffffe4, 0xfffffda0
+call 106, 0
# Move to retail spawn position
+mov 0xfffffe97, inl[:WS_TARGET_X]
+mov 0xffffffe4, inl[:WS_TARGET_Y]
+mov 0xfffffdb8, inl[:WS_TARGET_Z]
+mov 2, inl[:WS_ANIMATION_0]
+mov 6, inl[:WS_SPEED]
+call 105, inl[:WS_TARGET_X], inl[:WS_TARGET_Y], inl[:WS_TARGET_Z]
+gosub inl[:WS_MOVE]
# Face well
+mov 0xfffffe84, inl[:WS_TARGET_X]
+mov 0xfffffd44, inl[:WS_TARGET_Z]
+mov 0, inl[:WS_ANIMATION_0]
+mov 10, inl[:WS_FRAMES]
+gosub inl[:WS_ROTATE]
# Kicking animation
+call 97, 5
 mov 0x10, stor[24]
 gosub inl[:LABEL_0]
-mov 0xa, stor[24]
-wait stor[24]
-call 106, 0x0
# Face destination
+mov 0xffffff79, inl[:WS_TARGET_X]
+mov 0xfffffd15, inl[:WS_TARGET_Z]
+mov 0, inl[:WS_ANIMATION_0]
+mov 10, inl[:WS_FRAMES]
+gosub inl[:WS_ROTATE]
 mov 0xffffff79, stor[24]
 mov 0xffffffe4, stor[25]
 mov 0xfffffd15, stor[26]
