--- original
+++ modified
@@ -161,8 +161,6 @@
 call SMap::scriptSetModelPosition, stor[0], 0xffffff00, 0x0, 0x34b ; scriptIndex, x, y, z
 call SMap::scriptSetModelRotate, stor[0], 0x0, 0x0, 0x0 ; scriptIndex, x, y, z
 gosub inl[:LABEL_14] ; addr
-call SMap::scriptEnableSobjFlatLight, stor[0], 0x80, 0x80, 0x80 ; sobjIndex, r, g, b
-call SMap::scriptScaleUniform, stor[0], 0x1400 ; sobjIndex, scale
 return
 
 ; SUBROUTINE
