--- original
+++ modified
@@ -397,10 +397,13 @@
 LABEL_20:
 call 102, stor[0], stor[8], stor[9], stor[10]
 jmp_cmp ==, stor[8], 0xffffff83, inl[:LABEL_26]
+jmp_cmp ==, stor[8], 0xffffff82, inl[:LABEL_26]
 jmp_cmp ==, stor[8], 0xfffffeee, inl[:LABEL_27]
+jmp_cmp ==, stor[8], 0xfffffeed, inl[:LABEL_27]
 jmp_cmp ==, stor[8], 0x4, inl[:LABEL_28]
 jmp_cmp ==, stor[8], 0xf4, inl[:LABEL_29]
 jmp_cmp ==, stor[8], 0x54, inl[:LABEL_31]
+jmp_cmp ==, stor[8], 0x55, inl[:LABEL_31]
 jmp_cmp ==, stor[8], 0xb0, inl[:LABEL_30]
 jmp_cmp ==, stor[8], 0x6c, inl[:LABEL_32]
 LABEL_21:
