--- original
+++ modified
@@ -409,10 +409,13 @@
 LABEL_21:
 call SMap::scriptReadModelPosition, stor[0], stor[8], stor[9], stor[10] ; scriptIndex, x, y, z
 jmp_cmp ==, stor[8], 0xffffff83, inl[:LABEL_23] ; operand, left, right, addr
+jmp_cmp ==, stor[8], 0xffffff82, inl[:LABEL_23] ; operand, left, right, addr
 jmp_cmp ==, stor[8], 0xfffffeee, inl[:LABEL_24] ; operand, left, right, addr
+jmp_cmp ==, stor[8], 0xfffffeed, inl[:LABEL_24] ; operand, left, right, addr
 jmp_cmp ==, stor[8], 0x4, inl[:LABEL_25] ; operand, left, right, addr
 jmp_cmp ==, stor[8], 0xf4, inl[:LABEL_26] ; operand, left, right, addr
 jmp_cmp ==, stor[8], 0x54, inl[:LABEL_27] ; operand, left, right, addr
+jmp_cmp ==, stor[8], 0x55, inl[:LABEL_27] ; operand, left, right, addr
 jmp_cmp ==, stor[8], 0xb0, inl[:LABEL_28] ; operand, left, right, addr
 jmp_cmp ==, stor[8], 0x6c, inl[:LABEL_29] ; operand, left, right, addr
 LABEL_22:
