--- original
+++ modified
@@ -134,6 +134,7 @@
 jmp_cmp ==, 0x0, stor[24], inl[:LABEL_25] ; operand, left, right, addr
 call SMap::scriptReadModelPosition, var[64][0], stor[8], stor[9], stor[10] ; scriptIndex, x, y, z
 jmp_cmp ==, stor[8], 0xf0, inl[:LABEL_26] ; operand, left, right, addr
+jmp_cmp ==, stor[8], 0xf1, inl[:LABEL_26] ; operand, left, right, addr
 jmp_cmp ==, stor[8], 0xffffff3c, inl[:LABEL_27] ; operand, left, right, addr
 jmp_cmp ==, stor[8], 0xfffffef5, inl[:LABEL_28] ; operand, left, right, addr
 jmp_cmp ==, stor[8], 0x6f, inl[:LABEL_29] ; operand, left, right, addr
