--- original
+++ modified
@@ -1473,11 +1473,13 @@
 call 588, stor[18], 0x19
 call 238, 0x2, stor[17], 0x1, 0x0, 0x0, 0x0, 0x0
 gosub inl[:LABEL_26]
+jmp_cmp ==, stor[28], 0xffffffff, inl[:SKIP_0]
 call 600, stor[18]
 mov stor[27], stor[stor[stor[0], 18], 27]
 mov stor[28], stor[stor[stor[0], 18], 28]
 call 618, stor[18], inl[:LABEL_31]
 yield
+SKIP_0:
 call 600, stor[18]
 mov stor[29], stor[stor[stor[0], 18], 29]
 mov 0x2, stor[stor[stor[0], 18], 23]
@@ -1489,10 +1491,12 @@
 call 618, stor[18], inl[:LABEL_31]
 yield
 gosub inl[:LABEL_25]
+jmp_cmp ==, stor[28], 0xffffffff, inl[:SKIP_1]
 call 600, stor[18]
 mov stor[27], stor[stor[stor[0], 18], 27]
 mov stor[28], stor[stor[stor[0], 18], 28]
 call 618, stor[18], inl[:LABEL_31]
+SKIP_1:
 mov 0x2e, stor[23]
 wait stor[23]
 call 239, 0x2, stor[17], 0x1, 0x3
