--- original
+++ modified
@@ -196,14 +196,13 @@
 
 ; SUBROUTINE
 JMP_2a0_1:
-call SMap::scriptSelfLoadSobjModelAndAnimation, 0x8 ; objectIndex
+call SMap::scriptSelfLoadSobjModelAndAnimation, 0x2 ; objectIndex
 call SMap::scriptSelfSetSobjHidden, 0x0 ; value
-mov 0xa00, stor[24] ; source, dest
-call SMap::scriptSetAmbientColour, stor[0], stor[24], stor[24], stor[24] ; sobjIndex, r, g, b
-mov 0x1500, stor[24] ; source, dest
-mov 0x1c00, stor[25] ; source, dest
-call SMap::scriptScaleXyz, stor[0], stor[24], stor[25], stor[24] ; sobjIndex, x, y, z
 call SMap::scriptSetModelPosition, stor[0], 0xffffff26, 0xffffff6f, 0xffffffc2 ; scriptIndex, x, y, z
+call SMap::scriptSetModelRotate, stor[0], 0x0, 0xdc0, 0x0 ; scriptIndex, x, y, z
+call SMap::scriptSelfToggleAnimationDisabled, 0x1 ; disabled
+wait 0x6 ; frames
+call SMap::scriptSelfToggleAnimationDisabled, 0x0 ; disabled
 return
 
 ; SUBROUTINE
