--- original
+++ modified
@@ -1723,10 +1723,14 @@
 call 545, stor[0], stor[18], 0x0, 0xfffffc18, 0x0
 mov 0xa, stor[9]
 wait stor[9]
+jmp_cmp ==, stor[20], 0xffffffff, inl[:SKIP_P2]
 call 597, stor[0], stor[20], 0xa, 0x0, 0xfffffc18, 0x0
+SKIP_P2:
 mov 0xa, stor[9]
 wait stor[9]
+jmp_cmp ==, stor[19], 0xffffffff, inl[:SKIP_P1]
 call 597, stor[0], stor[19], 0x14, 0x0, 0xfffffc18, 0x0
+SKIP_P1:
 mov 0x14, stor[9]
 wait stor[9]
 LABEL_22:
@@ -1974,7 +1978,7 @@
 mov var[34][stor[9]], stor[10]
 call 138, stor[10], 0x0
 incr stor[9]
-jmp_cmp !=, stor[9], 0x3, inl[:LABEL_32]
+jmp_cmp !=, stor[9], var[35], inl[:LABEL_32]
 return
 LABEL_33:
 mov 0x0, stor[9]
@@ -1982,7 +1986,7 @@
 mov var[34][stor[9]], stor[10]
 call 138, stor[10], 0x1
 incr stor[9]
-jmp_cmp !=, stor[9], 0x3, inl[:LABEL_34]
+jmp_cmp !=, stor[9], var[35], inl[:LABEL_34]
 return
 LABEL_35:
 mov var[36][1], stor[10]
