--- original
+++ modified
@@ -2798,7 +2798,7 @@
 call 371, stor[0], stor[8], 0x0
 mov var[128 + stor[8]][3], stor[28]
 call 167, stor[28], stor[10]
-jmp_cmp !=, 0x0, stor[10], inl[:LABEL_194]
+jmp_cmp !=, stor[28], var[34][0], inl[:LABEL_194]
 or 0x2, var[45][31]
 or 0x4, var[45][31]
 or 0x20, stor[7]
@@ -2821,7 +2821,7 @@
 call 371, stor[0], stor[8], 0x0
 mov var[128 + stor[8]][3], stor[28]
 call 167, stor[28], stor[10]
-jmp_cmp !=, 0x0, stor[10], inl[:LABEL_195]
+jmp_cmp !=, stor[28], var[34][0], inl[:LABEL_195]
 or 0x2, var[45][31]
 or 0x4, var[45][31]
 or 0x20, stor[7]
@@ -3093,8 +3093,12 @@
 call 148, stor[26], 0xffffffff, 0x14, 0xffffe0c0, 0x0, 0x0
 mov 0x1, stor[23]
 wait stor[23]
+jmp_cmp ==, stor[24], 0xffffffff, inl[:SKIP_0]
 call 138, stor[24], 0x0
+SKIP_0:
+jmp_cmp ==, stor[25], 0xffffffff, inl[:SKIP_1]
 call 138, stor[25], 0x0
+SKIP_1:
 call 6, 0x4, 0x2
 mov 0x3, stor[23]
 wait stor[23]
