--- original
+++ modified
@@ -215,7 +215,7 @@
 call 371, stor[28], stor[10], 0x0
 mov stor[8], var[128 + stor[10]][5]
 mov 0x3, var[128 + stor[10]][0]
-call 168, stor[30], stor[8], ITEM_ID
+call 168, stor[30], reg[0], ITEM_ID
 mov stor[8], var[45][119]
 call 496, stor[30]
 call 508, stor[30], stor[28], stor[9]
@@ -223,8 +223,6 @@
 call 495, stor[30], stor[28], stor[8], stor[8], stor[10]
 mov stor[10], var[45][119]
 jmp_cmp ==, 0x0, stor[8], inl[:LABEL_1]
-call 169, stor[30], ITEM_PERCENTAGE, stor[10]
-jmp_cmp !=, 0x0, stor[10], inl[:LABEL_1]
 call 169, stor[30], STATUS, stor[10]
 jmp_cmp !&, 0x8, stor[10], inl[:LABEL_1]
 shr 0x1, stor[8]
@@ -1248,6 +1246,7 @@
 mov 0xcc, stor[8]
 gosub inl[:LABEL_5]
 call 592, stor[30], stor[28]
+mov id[lod:sachet], reg[0]
 mov 0x30, stor[29]
 call 236, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0
 call 746, stor[18], stor[28], 0xfff03, 0x8, 0x180, 0x3, 0x100, 0x4184000, 0x0
@@ -1267,6 +1266,7 @@
 mov stor[29], stor[8]
 gosub inl[:LABEL_0]
 fork_reenter stor[28], 0x2, 0xa
+mov 0xf0, var[45][230]
 mov 0xa, stor[23]
 wait stor[23]
 gosub inl[:LABEL_3]
